src/waveform/renderers/allshader/waveformrendererendoftrack.cpp

```diff
@@ -70,39 +81,32 @@ void WaveformRendererEndOfTrack::preprocess() {
     const double criticalIntensity = (remainingTimeTriggerSeconds - remainingTime) /
             remainingTimeTriggerSeconds;
 
-    const double alpha = criticalIntensity * blinkIntensity;
+    const double alpha = std::min(1.0, std::max(0.0, criticalIntensity * blinkIntensity));
```

Comment:
[`std::clamp`](https://en.cppreference.com/w/cpp/algorithm/clamp)

Reply:
Why didn’t I know this one 😭